Add a new knob to separately enable the autotuning in Triton.

nmacchioni opened the pull request:
Summary: separate triton pointwise autotune from matmul autotune, work done by ckluk

Test Plan: sandcastle + CI

Differential Revision: D43955699



cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

bertmaher reviewed (commented):
I'm a little worried our autotune configs are getting complex, and kind of opaque.  It seems like there are a few places we autotune that we want to control independently:
- pw/reduction: 3 levels
  - none
  - some: `autotune_pointwise` (default)
  - a lot: `max_autotune` (before this diff)
- gemm: 2 levels
  - none
  - a lot: `max_autotune`

I kind of think that we want is:
- more granularity on pw/reduction (`autotune_pointwise_moar`? (not serious))
- a separate switch for gemm since it's pretty expensive to do in prod (`autotune_matmul`)
- `max_autotune` to keep its current behavior of turning everything up to 11

nmacchioni commented:
> I'm a little worried our autotune configs are getting complex, and kind of opaque.  It seems like there are a few places we autotune that we want to control independently:
> 
> - pw/reduction: 3 levels
> 
>   - none
> 
>   - some: `autotune_pointwise` (default)
> 
>   - a lot: `max_autotune` (before this diff)
> 
> - gemm: 2 levels
> 
>   - none
> 
>   - a lot: `max_autotune`
> 
> 
> 
> I kind of think that we want is:
> 
> - more granularity on pw/reduction (`autotune_pointwise_moar`? (not serious))
> 
> - a separate switch for gemm since it's pretty expensive to do in prod (`autotune_matmul`)
> 
> - `max_autotune` to keep its current behavior of turning everything up to 11

I agree. What do you think about adding `max_autotune_pointwise` and `max_autotune_gemm` to differentiate the two, and keep `max_autotune` as an overall switch that sets both new options true?

ngimel commented:
That seems reasonable to me. @bertmaher?

nmacchioni committed:
Add a new knob to separately enable the autotuning in Triton. (#96440)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/96440

separate triton pointwise autotune from matmul autotune, work done by ckluk

Test Plan: sandcastle + CI

Reviewed By: jansel

Differential Revision: D43955699

fbshipit-source-id: 2a4212193fff06b5f55d70346597cf0e489975ec
